tests/blackbox/stratisd_cert.py

```diff
@@ -345,6 +345,50 @@ def test_pool_add_cache_permissions(self):
             StratisDbus.pool_add_cache, [pool_path, StratisCertify.DISKS[2:3]], True
         )
 
+    def test_pool_create_after_cache(self):
+        """
+        Test creating existing pool after cache was added
+        """
+        pool_name = p_n()
+        pool_path = make_test_pool(pool_name, StratisCertify.DISKS[0:1])
+
+        self._unittest_command(
+            StratisDbus.pool_init_cache(pool_path, StratisCertify.DISKS[1:2]),
+            dbus.UInt16(0),
+        )
+        self._unittest_command(
+            StratisDbus.pool_create(pool_name, StratisCertify.DISKS[0:1], None),
+            dbus.UInt16(0),
+        )
+
+    def test_pool_create_with_cache(self):
```

Comment:
This test is testing something different. I'm not sure if this is related.

Reply:
@mulkieran and I discussed adding some additional tests such as these

Reply:
Previously the set of existing devices was calculated as all the devices. So, if the create pool call specified the cache and data devices that were already in the existing pool the Identity constructor would be returned. This looks to me like a valid test of the change. @lleshchi One thing that you can try is to run your tests with the change in https://github.com/stratis-storage/stratisd/pull/2151 removed. If this test now fails, that indicates that the test is valuable.